tests/SpecTests/CrudSpecTest.php

```diff
@@ -19,11 +19,6 @@ class CrudSpecTest extends FunctionalTestCase
         'aggregate-merge: Aggregate with $merge and majority readConcern' => 'PHPLIB-438',
         'aggregate-merge: Aggregate with $merge and local readConcern' => 'PHPLIB-438',
         'aggregate-merge: Aggregate with $merge and available readConcern' => 'PHPLIB-438',
-        'aggregate-out-readConcern: readConcern majority with out stage' => 'PHPLIB-431',
```

Comment:
While working on these tests I realised that neither the `aggregate-out-readConcern` nor `aggregate-merge` spec tests cover running database-level aggregations with these options. I was wondering if it would make sense to add such tests to the spec to ensure consistent behaviour of the `db.aggregate` and `collection.aggregate` helpers?

Reply:
Since neither of the [DB-level pipeline operations](https://docs.mongodb.com/manual/reference/operator/aggregation-pipeline/#db-aggregate-stages) receive data from a collection, I'm not actually sure how read concern would affect the command. I could imagine it being ignored or prompting an server error. This would be worth asking in `#driver-devs` or, if necessary, `#server-query`. So long as it doesn't produce a server error, I don't see the harm in reproducing spec tests for it, since they could at least use APM.

I see no reason we cannot test `$merge`, though. And since it was introduced _after_ DB-level aggregations, I think you could just add a `db.aggregate()` case directly to the existing `$merge` spec test file since the `runOn` clause will apply.

Feel free to log a SPEC ticket (or two) for these and take it on. It would be a good learning experience for submitting a spec test PR.

Reply:
A quick test showed that running a DB-level pipeline (e.g. `$listLocalSessions`) with a read concern other than `local` will result in an error from the database. I'll put the information into a spec ticket once I've got the necessary permissions.